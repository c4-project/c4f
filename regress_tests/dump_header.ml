(* This file is part of c4f.

   Copyright (c) 2018-2022 C4 Project

   c4t itself is licensed under the MIT License. See the LICENSE file in the
   project root for more information.

   Parts of c4t are based on code from the Herdtools7 project
   (https://github.com/herd/herdtools7) : see the LICENSE.herd file in the
   project root for more information. *)

(** Tests the output of the 'act-c dump-header' command. *)

open Core

let dump_header_of_file ~file:(_ : Fpath.t) ~(path : Fpath.t) :
    unit Or_error.t =
  C4f_litmus_c.Header.Filters.run_dump
    (Plumbing.Input.of_fpath path)
    Plumbing.Output.stdout

let run (test_dir : Fpath.t) : unit Or_error.t =
  let full_dir = Fpath.(test_dir / "litmus" / "") in
  Common.regress_on_files "Dump-header" ~dir:full_dir ~ext:"litmus"
    ~f:dump_header_of_file

let command =
  Common.make_regress_command ~summary:"runs dump-header regressions" run

let () = Command_unix.run command
