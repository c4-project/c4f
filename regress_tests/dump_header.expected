# Dump-header tests

## deacon.c.litmus

```
{
  "name": "deacon2018_big",
  "locations": null,
  "init": { "x": 0, "y": 0 },
  "postcondition": "exists (1:r0 == 0 /\\ 2:r1 == 2)"
}
```
## fuzz_local_bools.c.litmus

```
{
  "name": "test_9",
  "locations": null,
  "init": {
    "game_11": 230,
    "clumsy_unicorn_10": -2,
    "gentle_corridor": -444497203,
    "zebra": 325,
    "y": 0,
    "x": 0
  },
  "postcondition": "forall\n((x == 1 /\\ (y == 1 /\\ 0:r0 == 0)) \\/\n ((x == 1 /\\ (y == 1 /\\ 0:r0 == 1)) \\/ (x == 1 /\\ (y == 2 /\\ 0:r0 == 0))))"
}
```
## iriw.c.litmus

```
{
  "name": "iriw",
  "locations": null,
  "init": { "x": 0, "y": 0 },
  "postcondition": "exists (2:r0 == 1 /\\ 2:r1 == 0 /\\ 3:r0 == 1 /\\ 3:r1 == 0)"
}
```
## iriw_fuzz_redundant.c.litmus

```
{
  "name": "iriw",
  "locations": [
    "MaJgQmON3yu", "ZgoV7uel9o", "VFTQNWN", "m02M5s", "ne0J8wrPQam", "x", "y"
  ],
  "init": {
    "MaJgQmON3yu": 2728,
    "ZgoV7uel9o": 1073,
    "VFTQNWN": -13,
    "m02M5s": -64856,
    "ne0J8wrPQam": -12197004,
    "x": 0,
    "y": 0
  },
  "postcondition": "exists (2:r0 == 1 /\\ 2:r1 == 0 /\\ 3:r0 == 1 /\\ 3:r1 == 0)"
}
```
## memalloy_cmpxchg.c.litmus

```
{
  "name": "test_7",
  "locations": null,
  "init": { "x": 0 },
  "postcondition": "exists (0:r0 == 0 /\\ x == 2)"
}
```
## memalloy_na_int.c.litmus

```
{
  "name": "test_0",
  "locations": null,
  "init": { "x": 0 },
  "postcondition": "exists (0:r0 == 0 /\\ x == 1)"
}
```
## nonzero_local.c.litmus

```
{
  "name": "test_8",
  "locations": null,
  "init": { "x": 0 },
  "postcondition": "forall\n((0:r0 == 0 /\\ 1:r0 == 0 /\\ x == 1) \\/ (0:r0 == 0 /\\ 1:r0 == 0 /\\ x == 2))"
}
```
## rwc.c.litmus

```
{
  "name": "rwc",
  "locations": null,
  "init": { "x": 0, "y": 0 },
  "postcondition": "exists (1:r0 == 1 /\\ 1:r1 == 0 /\\ 2:r0 == 0)"
}
```

Ran 8 test(s).
